trin-core/src/types/accumulator.rs

```diff
@@ -0,0 +1,203 @@
+        let block_2_rlp = hex::decode("f90218a088e96d4537bea4d9c05d12549907b32561d3bf31f45aae734cdc119f13406cb6a01dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d4934794dd2f1e6e498202e86d8f5442af596580a4f03c2ca04943d941637411107494da9ec8bc04359d731bfd08b72b4d0edcbd4cd2ecb341a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421b90100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008503ff00100002821388808455ba4241a0476574682f76312e302e302d30636463373634372f6c696e75782f676f312e34a02f0790c5aa31ab94195e1f6443d645af5b75c46c04fbf9911711198a0ce8fdda88b853fa261a86aa9e").unwrap();
+
+        let block_0_header: Header = rlp::decode(&block_0_rlp).unwrap();
+        let block_1_header: Header = rlp::decode(&block_1_rlp).unwrap();
+        let block_2_header: Header = rlp::decode(&block_2_rlp).unwrap();
+        let headers = vec![block_0_header, block_1_header, block_2_header];
+
+        // test values sourced from: https://github.com/status-im/nimbus-eth1/blob/master/fluffy/tests/test_accumulator.nim
+        let hash_tree_roots = vec![
+            hex::decode("b629833240bb2f5eabfb5245be63d730ca4ed30d6a418340ca476e7c1f1d98c0")
+                .unwrap(),
+            hex::decode("00cbebed829e1babb93f2300bebe7905a98cb86993c7fc09bb5b04626fd91ae5")
+                .unwrap(),
+            hex::decode("88cce8439ebc0c1d007177ffb6831c15c07b4361984cc52235b6fd728434f0c7")
+                .unwrap(),
+        ];
+        let mut master_accumulator = MasterAccumulator::new();
+        for (index, header) in headers.iter().enumerate() {
```

Comment:
*nit* maybe the more natural representation is to `zip(hash_tree_roots)` with `headers`, instead of `enumerate`.